src/ja/piccoma/src/eu/kanade/tachiyomi/extension/ja/piccoma/Piccoma.kt

```diff
@@ -0,0 +1,245 @@
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.firstInstance
+import keiyoushi.utils.parseAs
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import rx.Observable
+
+class Piccoma : HttpSource() {
+    override val name = "Piccoma"
+    override val baseUrl = "https://piccoma.com"
+    override val lang = "ja"
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient.newBuilder()
+        .addInterceptor(ImageInterceptor())
+        .build()
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/web/ranking/K/P/0", headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select("section.PCM-productRanking li > a").map { element ->
+            SManga.create().apply {
+                url = element.attr("href")
+                title = element.selectFirst(".PCM-rankingProduct_title p")!!.text()
+                element.selectFirst("img.js_lazy")?.absUrl("data-original")?.let { thumbnail_url = it }
+            }
+        }
+        return MangasPage(mangas, false)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$baseUrl/web/weekday/product/list".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select("li a:has(div.PCOM-prdList_info)").map { element ->
+            SManga.create().apply {
+                url = element.attr("href")
+                title = element.selectFirst(".PCOM-prdList_title span")!!.text()
+                element.selectFirst("img")?.absUrl("src")?.let { thumbnail_url = it }
+            }
+        }
+        val hasNextPage = document.select("#js_nextPage").isNotEmpty()
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            val url = "$baseUrl/web/search/result_ajax/list".toHttpUrl().newBuilder()
+                .addQueryParameter("word", query)
+                .addQueryParameter("page", page.toString())
+                .addQueryParameter("tab_type", "T")
+                .build()
+            return GET(url, super.headers.newBuilder().add("x-requested-with", "XMLHttpRequest").build())
+        }
+        val rankingPath = filters.firstInstance<RankingFilter>().toUriPart()
+        return GET("$baseUrl/web/ranking/$rankingPath", headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        if (response.request.url.toString().contains("/ranking/")) {
```

Comment:
Unless you're specifically excluding matching `ranking` in the last segment:
```suggestion
        if (response.request.url.pathSegments.contains("ranking")) {
```